### PR TITLE
Robustify against log files with corrupt messages

### PR DESCRIPTION
**Pull request by ndepal**
I have noticed that some ulog files fail to be opened by pyulog (e.g. with `ulog_info`, or when using pyulog as a library), but PlotJuggler is able to open that same file without any apparent issues.

pyulog throws the following error for such logs:
```Traceback (most recent call last):
  File "/home/nicolas/.local/bin/ulog_info", line 9, in <module>
    load_entry_point('pyulog', 'console_scripts', 'ulog_info')()
  File "/home/nicolas/src/pyulog/pyulog/info.py", line 85, in main
    ulog = ULog(ulog_file_name, None, disable_str_exceptions)
  File "/home/nicolas/src/pyulog/pyulog/core.py", line 132, in __init__
    self._load_file(log_file, message_name_filter_list)
  File "/home/nicolas/src/pyulog/pyulog/core.py", line 493, in _load_file
    self._read_file_data(message_name_filter_list)
  File "/home/nicolas/src/pyulog/pyulog/core.py", line 664, in _read_file_data
    msg_info = self._MessageInfo(data, header, is_info_multiple=True)
  File "/home/nicolas/src/pyulog/pyulog/core.py", line 281, in __init__
    self.key = type_key_split[1]
IndexError: list index out of range
```

This PR fixes the issue by first checking if `type_key_split` has the required size. If not, that message is skipped.
With this, `ulog_info` runs without errors.

Here is an example log where I have the problem: https://drive.google.com/file/d/1YpUzW61YnwS2KmNPbdoorw8Klu2oCcDo/view?usp=sharing

### CODE REVIEW

**Comment by ndepal** [comment]
Makes sense.

Like this?

I'll squash the commits if this is good.

File: pyulog/core.py [hunk 1]
```diff
@@ -657,62 +663,68 @@ def _read_file_data(self, message_name_filter_list, read_until=None):
...
+                            if self._has_sync:
+                                self._find_sync()
+                        else:
+                            # seek back msg_size to look for sync sequence in payload
+                            if self._has_sync:
+                                self._find_sync(header.msg_size)
+
+                except IndexError:
+                    if not self._file_corrupt:
+                        print("File corruption detected while reading file definitions!")
```
Comment (bkueng) [inline pyulog/core.py:726]: I guess you meant:
```suggestion
                        print("File corruption detected while reading file data!")
```

**Comment by ndepal** [comment]
I fixed the formatting issues. I think the other build failures are not my fault?

**Comment by ndepal** [comment]
Thanks for merging! Are there any plans for a new release soon?

**Comment by bkueng** [comment]
I usually collect a couple of things before triggering a release. But then again there are rare changes so I can do one now if you wish.

**Comment by ndepal** [comment]
It would make my life easier (I'm pulling this into a docker container). But I don't want to inconvenience you.

**Comment by bkueng** [comment]
Pushed as 0.8.0: https://pypi.org/project/pyulog/

**Comment by ndepal** [comment]
Awesome, thanks!

**Pull request merged**